src/transport/SessionManager.cpp

```diff
@@ -430,7 +430,12 @@ void SessionManager::OnMessageReceived(const PeerAddress & peerAddress, System::
     CHIP_TRACE_PREPARED_MESSAGE_RECEIVED(&peerAddress, &msg);
     PacketHeader packetHeader;
 
-    ReturnOnFailure(packetHeader.DecodeAndConsume(msg));
+    CHIP_ERROR err = packetHeader.DecodeAndConsume(msg);
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(Inet, "Failed to decode packet header: %" CHIP_ERROR_FORMAT, err.Format());
+        return;
+    }
```

Comment:
This can lead to garbage data attacks where log buffers are filled with those messages. If blocking logging is used to a UART, this can lead to priority inversions. This is very risky. PLEASE find a way to make this conditional

Reply:
I'd think on production firmware, actual emission of logging over UART will be disabled no?

Reply:
You would be extremely surprised how often it's not the case. The point about overflowing in-memory log buffer when there's a lot of crud on the wire is also true regardless of blocking logging or not

Reply:
Realistically this is more likely path for DOS attack I guess.
Are we explicitly trying to prevent DOS from the internal network (given that chip is internal network)? I am not sure if we have restrictions anywhere else ... I imagine mdns probably logs as well. Or if not, I could replay seen packets, they will pass this check (decode is ok) and then still log (see CHIP_ERROR_DUPLICATE_MESSAGE_RECEIVED log below this for example).